backend/service/service_proposal.go

```diff
@@ -93,6 +93,151 @@ func (service *ProposalService) QueryDepositAndVotingProposalList() []vo.Proposa
+	}
+
+	tx, err := document.CommonTx{}.QueryProposalInitAmountTxById(id)
+	if err != nil {
+		logger.Error("QueryProposalInitAmountTxById have error", logger.String("err", err.Error()))
+		panic(types.CodeNotFound)
+	}
+	proposal.InitialDeposit = vo.Coin{
+		Denom:  tx.Amount[0].Denom,
+		Amount: tx.Amount[0].Amount,
+	}
+
+	l := vo.Level{}
+	name, err := lcd.GetProposalLevelByType(data.Type)
+	if err != nil {
+		logger.Error("get proposal level by type", logger.String("err", err.Error()), logger.String("param", data.Type))
+	}
+	l.Name = name
+
+	coinAsDoc, err := lcd.GetMinDepositByProposalType(data.Type)
+	if err != nil {
+		logger.Error("get min deposit", logger.String("err", err.Error()), logger.String("param", data.Type))
+	}
+	l.GovParam = vo.GovParam{
+		MinDeposit: vo.Coin{
+			Amount: coinAsDoc.Amount,
+			Denom:  coinAsDoc.Denom,
+		},
+	}
+	proposal.Level = l
+
+	if len(data.TotalDeposit) > 0 {
+		proposal.TotalDeposit.Amount = data.TotalDeposit[0].Amount
+		proposal.TotalDeposit.Denom = data.TotalDeposit[0].Denom
+	}
+
+	return proposal
+}
+
+func (service *ProposalService) QueryVoting(id int) vo.ProposalNewStyle {
+	data, err := document.Proposal{}.QueryProposalById(id)
+	if err != nil {
+		logger.Error("QueryProposalById have error", logger.String("err", err.Error()))
+		panic(types.CodeNotFound)
+	}
+
+	systemVotingPower, err := service.GetSystemVotingPower()
+	if err != nil {
+		logger.Error("get systemVotingPower fail", logger.String("err", err.Error()))
+	}
+
+	tmpVoteArr := make([]vo.VoteWithVoterInfo, 0, len(data.Votes))
+	finalVotes := vo.FinalVotes{}
+
+	if data.Status == document.ProposalStatusPassed || data.Status == document.ProposalStatusRejected {
+		finalVotes = vo.FinalVotes{
+			Yes:        data.FinalVotes.Yes,
+			No:         data.FinalVotes.No,
+			NoWithVeto: data.FinalVotes.NoWithVeto,
+			Abstain:    data.FinalVotes.Abstain,
+		}
+	}
+
+	if data.Status == document.ProposalStatusVoting {
+		allBondedValidators, err := document.Validator{}.GetBondedValidatorsSharesTokens()
+		if err != nil {
+			logger.Error("query allBondedValidators", logger.String("err", err.Error()))
+		}
+		curValidators := make(map[string]*vo.ValidatorGovInfo, len(allBondedValidators))
+
+		for _, v := range allBondedValidators {
+			delegatorShares, _ := utils.ParseStringToFloat(v.DelegatorShares)
+			tokens, _ := utils.ParseStringToFloat(v.Tokens)
+			curValidators[v.OperatorAddress] = &vo.ValidatorGovInfo{
+				Address:   v.OperatorAddress,
+				DelShares: delegatorShares,
+				Tokens:    tokens,
+			}
+		}
+		curDelegators := make(map[string]*vo.DelegatorGovInfo, len(data.Votes))
+
+		for _, v := range data.Votes {
+			getDelegationsByVoter(curValidators, curDelegators, v)
+		}
+		computedValidatorsPower(curValidators, curDelegators)
+
+		for _, v := range curDelegators {
+			var voterVotingPower float64
+			voterVotingPower = v.DelVotingPower + v.ValVotingPower
+			tmpVote := vo.VoteWithVoterInfo{
```

Comment:
add field `DelVotingPower ` and `ValVotingPower ` for `VoteWithVoterInfo ` struct

Reply:
ok